src/RoslynAnalyzers/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -658,27 +662,29 @@ await checkHelpLinkAsync(helpLinkUri).ConfigureAwait(false))
 
                 return;
 
-                async Task<bool> checkHelpLinkAsync(string helpLink)
+                async Task<(HttpStatusCode, string? responseContent)> checkHelpLinkAsync(string helpLink)
                 {
                     try
                     {
                         if (!Uri.TryCreate(helpLink, UriKind.Absolute, out var uri))
                         {
-                            return false;
+                            // TODO: why would we not fail this if the URI is invalid?
+                            return (HttpStatusCode.OK, null);
                         }
 
                         if (validateOffline)
                         {
-                            return true;
+                            // Just pretend the request worked fine.
+                            return (HttpStatusCode.OK, null);
                         }
 
                         var request = new HttpRequestMessage(HttpMethod.Head, uri);
                         using var response = await httpClient.SendAsync(request).ConfigureAwait(false);
-                        return response?.StatusCode == HttpStatusCode.OK;
+                        return (response.StatusCode, await response.Content.ReadAsStringAsync().ConfigureAwait(false));
```

Comment:
Unsure if a HEAD request will ever have Content (the point is to not...) but will return it just in case it still comes for an error?